test/syncenginetestutils.cpp

```diff
@@ -568,7 +569,7 @@ QVector<FileInfo *> FakePutMultiFileReply::performMultiPart(FileInfo &remoteRoot
             // Assume that the file is filled with the same character
             fileInfo = remoteRootFileInfo.create(fileName, onePartBody.size(), onePartBody.at(0).toLatin1());
         }
-        fileInfo->lastModified = OCC::Utility::qDateTimeFromTime_t(request.rawHeader("x-oc-mtime").toLongLong());
+        fileInfo->lastModified = OCC::Utility::qDateTimeFromTime_t(modtime);
```

Comment:
that would differ from server behavior
modification time of the server side file should be what is in the header `x-oc-mtime`

Reply:
not really -- `request.rawHeader` only contains the headers of the HTTP request itself, not the multipart-part that's being uploaded (and `modtime` is set to the value from the multipart headers)

while cross-checking this I just saw that the header used for the `modtime` variable uses the `x-file-mtime` multipart header, which has a TODO set on the server side: https://github.com/nextcloud/server/blob/c029616ec05f363b28467bbdccd5ce806fd43212/apps/dav/lib/BulkUpload/BulkUploadPlugin.php#L61-L68